compiler/rustc_borrowck/src/dataflow.rs

```diff
@@ -461,34 +460,26 @@ impl<'a, 'tcx> Borrows<'a, 'tcx> {
         regioncx: &RegionInferenceContext<'tcx>,
         borrow_set: &'a BorrowSet<'tcx>,
     ) -> Self {
-        let mut borrows_out_of_scope_at_location =
-            calculate_borrows_out_of_scope_at_location(body, regioncx, borrow_set);
-
-        // The in-tree polonius analysis computes loans going out of scope using the set-of-loans
-        // model, and makes sure they're identical to the existing computation of the set-of-points
-        // model.
-        if tcx.sess.opts.unstable_opts.polonius.is_next_enabled() {
-            let mut polonius_prec = PoloniusOutOfScopePrecomputer::new(body, regioncx);
-            for (loan_idx, loan_data) in borrow_set.iter_enumerated() {
-                let issuing_region = loan_data.region;
-                let loan_issued_at = loan_data.reserve_location;
-
-                polonius_prec.precompute_loans_out_of_scope(
-                    loan_idx,
-                    issuing_region,
-                    loan_issued_at,
-                );
-            }
-
-            assert_eq!(
-                borrows_out_of_scope_at_location, polonius_prec.loans_out_of_scope_at_location,
-                "polonius loan scopes differ from NLL borrow scopes, for body {:?}",
-                body.span,
-            );
-
-            borrows_out_of_scope_at_location = polonius_prec.loans_out_of_scope_at_location;
-        }
+        let borrows_out_of_scope_at_location =
+            if !tcx.sess.opts.unstable_opts.polonius.is_next_enabled() {
+                calculate_borrows_out_of_scope_at_location(body, regioncx, borrow_set)
+            } else {
+                // The in-tree polonius analysis computes loans going out of scope using the
+                // set-of-loans model.
+                let mut polonius_prec = PoloniusOutOfScopePrecomputer::new(body, regioncx);
+                for (loan_idx, loan_data) in borrow_set.iter_enumerated() {
```

Comment:
I know this is preexisting, but personally I would prefer this to be inside `PoloniusOutOScopePrecomputer::new` (you could rename to `compute` and make it return `loans_out_of_scope_at_location`.

This could be done in a separate PR (before or after this PR lands).

Reply:
I can do that in the next PR, sure. The structure used to be identical to the NLL scope precomputer in this function, but that was since moved out to `calculate_borrows_out_of_scope_at_location`. I’ll try to keep the symmetry with NLLs and do what you suggest there as well if possible.

Reply:
I pushed a commit to do this